packages/core/src/core/fetch.ts

```diff
@@ -280,13 +286,38 @@ const fetchAndProcessMergeErrors = async (
 
     log.debug(`after merge there are ${processErrorsResult.keptElements.length} elements [errors=${
       mergeErrors.length}]`)
-    return { serviceElements: validServiceElements, processErrorsResult, updatedConfigs }
+    return {
+      serviceElements: validServiceElements,
+      processErrorsResult,
+      updatedConfigs,
+      adapterToParital,
+    }
   } catch (error) {
     getChangesEmitter.emit('failed')
     throw error
   }
 }
 
+export const getAdaptersFirstFetchPartial = (
+  elements: Element[],
+  adapterToParital: Record<string, boolean>,
+): string[] => {
+  const fetchedAdapaters = new Set<string>()
+  elements.forEach(e => {
+    fetchedAdapaters.add(e.elemID.adapter)
+  })
+
+  const firstPartialFetchAdapter: string[] = []
+  wu.entries(adapterToParital).forEach(
+    ([adapter, isPartial]) => {
+      if (isPartial && !fetchedAdapaters.has(adapter)) {
+        firstPartialFetchAdapter.push(adapter)
+      }
+    }
+  )
+  return firstPartialFetchAdapter
```

Comment:
using side-effect based flow here is much more complicated then functional IMO
```suggestion
  const adaptersWithElements = new Set(elements.map(e => e.elemID.adapter))
  const partiallyFetchedAdapters = _.pickBy(adapterToPartial, isPartial => isPartial) // btw - might be easier to just return a list of which adapters were partial instead of this object...
    return collections.set.difference(partiallyFetchedAdapters, adaptersWithElements) // defined in lowerdash
```